spec.go

```diff
@@ -94,6 +94,8 @@ type Spec struct {
 	// Each item in this list is run with a separate rootfs and cannot interact with other tests.
 	// Each [TestSpec] is run with a separate rootfs, asynchronously from other [TestSpec].
 	Tests []*TestSpec `yaml:"tests,omitempty" json:"tests,omitempty"`
+
+	Ext map[string]interface{} `yaml:"-" json:"-"`
```

Comment:
I wonder if we can do something different here and allow callers to unmarshal directly to their own data type?

Reply:
Seems difficult to deal with while also requiring fields to start with `x-`

Reply:
Actually I think this doesn't matter since the map value would be the callers data type, which the caller can assert.